src/site/artifacts/build.rs

```diff
@@ -0,0 +1,67 @@
+use crate::config::Config;
+use crate::errors::*;
+use crate::site::layout;
+use axohtml::{html, text, unsafe_text};
+use cargo_dist_schema::ArtifactKind;
+
+use super::fetch_manifest;
+use crate::config::artifacts::Artifacts;
+use crate::site::artifacts::cargo_dist;
+use crate::site::artifacts::package_managers;
+
+pub fn build_page(config: &Config) -> Result<()> {
+    let mut html = vec![];
+    let manifest = fetch_manifest(&config)?;
+
+    if let Some(Artifacts {
+        package_managers: Some(managers),
+        ..
+    }) = &config.artifacts
+    {
+        let mut list = vec![];
+
+        if let Some(Artifacts {
+            cargo_dist: Some(true),
+            ..
+        }) = &config.artifacts
+        {
+            for release in manifest.releases.iter() {
```

Comment:
it may help with code organization to break some of these loops out into their own functions?

Reply:
moved some stuff let me know thoughts

Reply:
nice!